Description: Fixed insecure permissions for /tmp/fastnetmon.dat
Author: Pavel Odinstsov <pavel.odintsov@gmail.com>
Last-Update: 2018-12-20
Applied-Upstream: https://github.com/pavel-odintsov/fastnetmon/commit/978289dd29622067b47396706f2252ad1eace633

--- a/src/fastnetmon.cpp
+++ b/src/fastnetmon.cpp
@@ -2397,6 +2397,9 @@ void print_screen_contents_into_file(std
     screen_data_file.open(cli_stats_file_path.c_str(), std::ios::trunc);
 
     if (screen_data_file.is_open()) {
+        // Set 660 permissions to file for security reasons
+        chmod(cli_stats_file_path.c_str(), S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH);
+
         screen_data_file << screen_data_stats_param;
         screen_data_file.close();
     } else {
